receiver/mysqlreceiver/scraper.go

```diff
@@ -128,10 +88,8 @@ func (m *mySQLScraper) scrape(context.Context) (pdata.Metrics, error) {
 		if k != "buffer_pool_size" {
 			continue
 		}
-		labels := pdata.NewAttributeMap()
 		if f, ok := m.parseFloat(k, v); ok {
-			labels.Insert(metadata.A.BufferPoolSize, pdata.NewAttributeValueString("total"))
-			addToDoubleMetric(bufferPoolSize, labels, f, now)
+			m.mb.RecordMysqlBufferPoolSizeDataPoint(now, f, "total")
```

Comment:
Aren't we recording different values with the same attribute here? I would expect `k` to be passed instead of `"total"` but not sure how it supposed to work

Reply:
I think you're right that there's a problem here, but not quite as described. 

I've written up #7511, which covers this and a few related problems. In short though, I think this particular data point should be its own metric w/ no attribute.

Reply:
I think it makes sense to handle that in a separate PR so this one can just be a switch to metrics builder w/o changes to functionality.

Reply:
Sounds good. Thanks